src/l1block.rs

```diff
@@ -136,12 +138,65 @@ impl L1BlockInfo {
         self.calldata_gas.unwrap().saturating_add(blob_gas).wrapping_div(TX_L1_FEE_PRECISION)
     }
 
+    fn calculate_tx_l1_cost_feynman(&self, input: &[u8], spec_id: ScrollSpecId) -> U256 {
+        // rollup_fee(tx) = compression_ratio(tx) * size(tx) * (component_exec + component_blob)
+        //
+        // - compression_ratio(tx): estimated compression ratio of RLP-encoded signed tx data,
+        // derived by plugging in this tx into the previous finalised batch. This gives us an idea
+        // as to how compressible is the zstd-encoding of the data in this tx.
+        //
+        // - size(tx): denotes the size of the RLP-encoded signed tx data.
```

Comment:
Just noting to be precise: For eip-2718 transactions, this includes the type byte that precedes the rlp-encoded transaction.